vars/skipStage.groovy

```diff
@@ -10,6 +13,34 @@
  * Method to return true or false if a commit pragma says to skip a stage
  */
 
+// Determine if this is a restarted job and if it succeeded on the
+// previous run.
+boolean already_passed() {
+    String restart_cause = 'org.jenkinsci.plugins.pipeline.modeldefinition.' +
+                   'causes.RestartDeclarativePipelineCause'
+
+    restarted = currentBuild.getBuildCauses().any { cause ->
+        cause._class == restart_cause }
+    if (!restarted) {
+        return false
+    }
+    String my_build = "/${env.BUILD_NUMBER}/"
+    String old_build = "/${env.BUILD_NUMBER.toInteger() - 1}/"
+
+    String old_job = env.BUILD_URL.replace(my_build, old_build)
+    String art_url = old_job + 'artifact/' + stageStatusFilename()
+
+    def response = httpRequest url: art_url,
+                               acceptType: 'TEXT_PLAIN',
+                               httpMode: 'GET',
+                               validResponseCodes: '100:599'
+
+    if (response.status == 200 && response.content == 'SUCCESS') {
+        return true
+    }
+    return false
```

Comment:
Couldn't you use the more idiomatic [Copy Artifact](https://plugins.jenkins.io/copyartifact/) plugin for this?

Reply:
I have not been able to get the CopyArtifact to work reliably with the Github multibranch pipeline jobs as a target.  The required syntax is not documented for that Plugin.

Reply:
Work at all or just reliably?  Like it works sometimes and not others?

Reply:
Current code now first tries to use copyArtifact which requires being enabled in the Jenkinsfile to work, and if copyArtifact is not able to find the artifact, it falls back to httpRequest to read the artifact, which allows this change to be used before the Jenkinsfiles are updated.

Also moves the artifact file into a separate directory in the artifacts.